unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/MyAccountHUD/MyAccountCardController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using DCL.Browser;
+using DCL.SettingsCommon;
+using System;
+
+namespace DCL.MyAccount
+{
+    public class MyAccountCardController
+    {
+        private const string OPEN_PASSPORT_SOURCE = "ProfileHUD";
+        private const string URL_TERMS_OF_USE = "https://decentraland.org/terms";
+        private const string URL_PRIVACY_POLICY = "https://decentraland.org/privacy";
+
+        private readonly IMyAccountCardComponentView view;
+        private readonly DataStore dataStore;
+        private readonly IUserProfileBridge userProfileBridge;
+        private readonly Settings settings;
+        private readonly IBrowserBridge browserBridge;
+
+        public MyAccountCardController(
+            IMyAccountCardComponentView view,
+            DataStore dataStore,
+            IUserProfileBridge userProfileBridge,
+            Settings settings,
+            IBrowserBridge browserBridge)
+        {
+            this.view = view;
+            this.dataStore = dataStore;
+            this.userProfileBridge = userProfileBridge;
+            this.settings = settings;
+            this.browserBridge = browserBridge;
+
+            view.OnPreviewProfileClicked += OnPreviewProfileClicked;
+            view.OnAccountSettingsClicked += OnAccountSettingsClicked;
+            view.OnSignOutClicked += OnSignOutClicked;
+            view.OnTermsOfServiceClicked += OnTermsOfServiceClicked;
+            view.OnPrivacyPolicyClicked += OnPrivacyPolicyClicked;
+
+            userProfileBridge.GetOwn().OnUpdate += OnOwnUserProfileUpdate;
+            if (!string.IsNullOrEmpty(userProfileBridge.GetOwn().userId))
+                OnOwnUserProfileUpdate(userProfileBridge.GetOwn());
```

Comment:
Why do we check for the userId to not be null if in OnOwnUserProfileUpdate we check the entire userProfile if it's null?

Reply:
The userProfile may be not null but the profile might not be initialized. I added a small variable to make the code verbose